/*
 * Copyright 2017-2019 PixelsDB.
 *
 * This file is part of Pixels.
 *
 * Pixels is free software: you can redistribute it and/or modify
 * it under the terms of the Affero GNU General Public License as
 * published by the Free Software Foundation, either version 3 of
 * the License, or (at your option) any later version.
 *
 * Pixels is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * Affero GNU General Public License for more details.
 *
 * You should have received a copy of the Affero GNU General Public
 * License along with Pixels.  If not, see
 * <https://www.gnu.org/licenses/>.
 */
syntax = "proto3";

option java_multiple_files = false;
option java_package = "io.pixelsdb.pixels.retina";
option java_outer_classname = "RetinaWriterProto";

package retina_writer.proto;

service RetinaWriterService {
  rpc Flush (FlushRequest) returns (FlushResponse);
}

message FlushRequest {
  string schemaName = 1;
  string tableName = 2;
  sint32 rgid = 3;
  string filePath = 4;
  uint64 pos = 5;
}

message FlushResponse {
  int32 errorCode = 1;
  uint64 pos = 2;
}

